back/KinoFund/BLL/Comments/CommentsManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Core.Dtos.Comments;
+using DAL.data;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BLL.Comments
+{
+    public class CommentsManager
+    {
+        private readonly MyContext _dbContext;
+
+        public CommentsManager(MyContext context)
+        {
+            _dbContext = context;
+        }
+
+        public async Task<GetAllComentsDTO> GetAllAsync(long movieId)
+        {
+
+            var comments = await _dbContext.Comments
+                .Include(i => i.User)
+                .Include(i => i.Movie)
+                .Where(x => x.MovieId == movieId)
+                .Select(c => new CommentDTO
+                {
+                    CommentId = c.CommentId,
+                    UserName = c.User.UserName,
+                    Text = c.Text,
+                    ParentCommentId = c.RefersToCommentId,
+                    SubComments = c.Movie.Comments.Where(x => x.RefersToCommentId == c.CommentId).Select(x => new SubCommentIdDTO
+                    { 
+                        SubCommentId = x.CommentId
+
+                        
+                    }).ToList()
+
+                }).ToListAsync();
+
+            
+
+            return new GetAllComentsDTO
+            {
+                Comments = comments,
```

Comment:
Do you see on the result of your method?
It should be as this
[{
   CommentId = 1,
   SubComments: [{
   CommentId=2,
   SubComments=[]
   }, {
   CommentId=3,
   SubComments=[]